core/src/main/java/com/graphhopper/routing/Dijkstra.java

```diff
@@ -68,19 +67,19 @@ public Path calcPath(int from, int to) {
     }
 
     protected void runAlgo() {
-        EdgeExplorer explorer = outEdgeExplorer;
         while (true) {
             visitedNodes++;
             if (isMaxVisitedNodesExceeded() || finished())
                 break;
 
             int currNode = currEdge.adjNode;
-            EdgeIterator iter = explorer.setBaseNode(currNode);
+            EdgeIterator iter = edgeExplorer.setBaseNode(currNode);
             while (iter.next()) {
                 if (!accept(iter, currEdge.edge))
                     continue;
 
-                double tmpWeight = weighting.calcWeight(iter, false, currEdge.edge) + currEdge.weight;
+                // todo: for #1776/#1835 move the access check into weighting
+                double tmpWeight = !outEdgeFilter.accept(iter) ? Double.POSITIVE_INFINITY : weighting.calcWeight(iter, false, currEdge.edge) + currEdge.weight;
```

Comment:
In #18350 we can remove outEdgeFilter here and strictly rely on the returned weight from `Weighting`